server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/SimpleSearchQueryConverter.java

```diff
@@ -347,4 +349,8 @@ private Object getStringOrDate(String s) {
     public String map(String attribute) {
         return hasText(attribute) ? mapper.mapToInternal(attribute) : attribute;
     }
+
+    public String getCustomPrefix() {
+        return (mapper instanceof JoinAttributeNameMapper joinAttributeNameMapper) ? joinAttributeNameMapper.getCustomPrefix() : UaaStringUtils.EMPTY_STRING;
```

Comment:
I'm not happy about the design requiring the use of "instanceof", though it's not as bad as the cast mentioned in the comment below.

Reply:
Ok, I am surprised, because "instance of" is a typical UAA pattern because we have so many interfaces and mixing then the objects, but there is a good news with java 17. instance of has received an improvement so that you dont have to check and cast an object but now you can do it with one statement - like I did.

Sonar findings about instance of, e.g. (we have here 191 smells): 
https://sonarcloud.io/project/issues?resolved=false&rules=java%3AS6201&id=cloudfoundry-identity-parent

Additional ref. https://howtodoinjava.com/java/basics/pattern-matching-instanceof/

Reply:
Sure, that would be an improvement. 

Just because instanceof is used in many places doesn't make it any better. :)